### PR TITLE
fix

### PR DESCRIPTION
**Pull request by zhuoshen617**

### CODE REVIEW

File: test/pymongo_test.py [hunk 1]
```diff
@@ -1,69 +0,0 @@
-from dotenv import load_dotenv
```
Comment (zhuoshen617) [inline test/pymongo_test.py:1]: delete this file. or make it gitignore

File: scheduler.py [hunk 2]
```diff
@@ -1,173 +0,0 @@
...
-        clean_word = word.strip()
-        if clean_word in TICKER_BLACKLIST:
-            continue
-        if clean_word in ticker_dict:
-            ticker_count[clean_word] = ticker_count.get(clean_word, 0) + 1
-
-
-# Helper function to determine if Yahoo Finance can provide the relevant information
-# If not, we skip this stock
-def found_all_fields(stock_info):
```
Comment (zhuoshen617) [inline scheduler.py:66]: has_all_finance_fields

File: scheduler.py [hunk 3]
```diff
@@ -1,173 +0,0 @@
...
-
-# Number of the most discussed stocks we save
-# This should be greater than 10, in case Yahoo Finance is not able to provide information for certain stocks
-TOTAL_NUMBER_OF_STOCKS = 20
-
-# A list of abbreviations that generally do not represent tickers.
-TICKER_BLACKLIST = ['A', 'ALL', 'ARE', 'AT', 'CEO', 'DD', 'EDIT', 'EV', 'FOR', 'ON', 'ONE', 'OR', 'RH', 'VERY']
-
-# Information we look for in Yahoo Finance
-YF_INFO_FIELDS = ['industry', 'previousClose', 'fiftyDayAverage', 'averageDailyVolume10Day']
```
Comment (zhuoshen617) [inline scheduler.py:19]: yahoo_finance_FIELDS

File: scheduler.py [hunk 4]
```diff
@@ -1,173 +0,0 @@
...
-        ranking_change = len(to_DB) - prev_stock_list.index(output_ticker)
-
-    return is_new, ranking_change
-
-
-# Packages the Top 10 tickers and related financial information into a list of dictionaries
-# also inserts this information into the database with time stamps
-def insert_to_db(sorted_ticker_dict, ticker_dict):
-    client = MongoClient(os.environ['DB_URL'])
-    collection = client['findMyGME']['stocks']
```
Comment (zhuoshen617) [inline scheduler.py:107]: db_collection

File: scheduler.py [hunk 5]
```diff
@@ -1,173 +0,0 @@
...
-        if field not in stock_info.keys():
-            return False
-
-    return True
-
-
-# Helper function to get the most recent Top 10 stock record from the database
-# Returns a list of ticker symbols in non-decreasing order
-def get_prev_stock_list(collection, now):
-    prev_entry = collection.find({'timeStamp': {'$lte': now}}).sort([('timeStamp', -1)]).limit(1)
```
Comment (zhuoshen617) [inline scheduler.py:77]: make now a local var.

Comment (zhuoshen617) [reply scheduler.py:77]: lastest_timestamp_entry = collection.find({'t

File: scheduler.py [hunk 6]
```diff
@@ -1,173 +0,0 @@
...
-
-    prev_time_stamp = prev_entry[0].get('timeStamp')
-    prev_stock_obj_list = collection.find({'timeStamp': prev_time_stamp})
```
Comment (zhuoshen617) [inline scheduler.py:80]: you don't have a sort, how is the list in non-decreasing order

File: scheduler.py [hunk 7]
```diff
@@ -1,173 +0,0 @@
...
-    prev_stock_list = []
-    for stock_obj in prev_stock_obj_list:
-        prev_stock_list.append(stock_obj.get('ticker'))
-
-    return prev_stock_list
-
-
-# Helper function to calculate ranking changes for the current Top 10 stocks
-# returns a tuple of whether this stock is new on the list and the ranking change (if applicable)
-def get_ranking_change(prev_stock_list, output_ticker, to_DB):
```
Comment (zhuoshen617) [inline scheduler.py:91]: def get_ranking_change(prev_stock_list, **ticker**, to_DB):

Comment (zhuoshen617) [reply scheduler.py:91]: pass in current rank, getting  current rank from len(to_DB) is a terrible idea.

File: scheduler.py [hunk 8]
```diff
@@ -1,173 +0,0 @@
...
-    is_new = False
```
Comment (zhuoshen617) [inline scheduler.py:92]:
```
is_new = False
ranking_change = 0
if ticker not in prev_stock_list:
    is_new = True
    return is_new, ranking_change

ranking_change = len(to_DB) - prev_stock_list.index(output_ticker)
return is_new, ranking_change
```

File: scheduler.py [hunk 9]
```diff
@@ -1,173 +0,0 @@
...
-            'rank': len(to_DB) + 1,
-            'ticker': output_ticker,
-            'name': ticker_dict[output_ticker],
-            'industry': stock_info.get('industry', ''),
-            'count': output_count,
-            'previousClose': stock_info.get('previousClose', ''),
-            'fiftyDayAverage': stock_info.get('fiftyDayAverage', ''),
-            'averageDailyVolume10Day': stock_info.get('averageDailyVolume10Day', ''),
-            'timeStamp': now,
-            'newOccur': ranking_change[0],
```
Comment (zhuoshen617) [inline scheduler.py:137]: why don't call it isNew?

File: scheduler.py [hunk 10]
```diff
@@ -1,173 +0,0 @@
...
-
-    for i in range(TOTAL_NUMBER_OF_STOCKS):
-        output_ticker = sorted_ticker_list[i]
-        output_count = sorted_ticker_dict[output_ticker]
-        stock_info = yf.Ticker(output_ticker).info
-
-        if not found_all_fields(stock_info):
-            continue
-
-        ranking_change = get_ranking_change(prev_stock_list, output_ticker, to_DB)
```
Comment (zhuoshen617) [inline scheduler.py:125]: isNew, ranking_change = get_ranking_change(prev_stock_list, output_ticker, to_DB)

File: scheduler.py (hunk 5 above)
Comment (LilyXueLi) [reply scheduler.py:77]: to discussed

File: scheduler.py (hunk 6 above)
Comment (LilyXueLi) [reply scheduler.py:80]: to be discussed

File: scheduler.py (hunk 7 above)
Comment (LilyXueLi) [reply scheduler.py:91]: to be discussed

File: scheduler.py (hunk 9 above)
Comment (LilyXueLi) [reply scheduler.py:137]: discussed

File: test/pymongo_test.py (hunk 1 above)
Comment (LilyXueLi) [reply test/pymongo_test.py:1]: to be discussed

**Pull request closed without merging**